core/include/drivers/clk.h

```diff
@@ -0,0 +1,287 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2021, Bootlin
+ */
+
+#ifndef CLK_H
+#define CLK_H
+
+#include <kernel/dt.h>
+#include <stdint.h>
+#include <sys/queue.h>
+
+#define CLK_NO_SCMI_ID	-1
+
+struct clk;
+
+/**
+ * struct clk_ops
+ *
+ * @enable: Enable the clock
+ * @disable: Disable the clock
+ * @is_enabled: Check if the clock is enabled
+ * @set_parent: Set the clock parent based on index
+ * @get_parent: Get the current parent index of the clock
+ * @set_rate: Set the clock rate
+ * @get_rate: Get the clock rate
+ * @get_rates_array: Get the supported clock rates as array
+ */
+struct clk_ops {
+	int (*enable)(struct clk *clk);
+	void (*disable)(struct clk *clk);
+	int (*is_enabled)(struct clk *clk);
+	int (*set_parent)(struct clk *clk, uint8_t index);
+	uint8_t (*get_parent)(struct clk *clk);
```

Comment:
Do you have real needs for these set_parent/get_parent handlers?

Reply:
In the current clock tree that I'm porting (at91), there is need to set and get the parent for USB for instance.
We could potentially set them statically but then we will loose the ability to change to new clock rates by using new parents.